src/lib/y2network/interface_config_builders/infiniband.rb

```diff
@@ -29,31 +28,22 @@ def initialize(config: nil)
         super(type: InterfaceType::INFINIBAND, config: config)
       end
 
-      # @return [String] ipoib mode configuration
-      attr_writer :ipoib_mode
+      # @param value [String] ipoib mode configuration
+      def ipoib_mode=(value)
+        value = "" if value == "default"
+        connection_config.ipoib_mode = IpoibMode.from_name(value)
+      end
 
       # Returns current value of infiniband mode
       #
       # @return [String] particular mode or "default" when not set
       def ipoib_mode
-        @ipoib_mode ||= if [nil, ""].include?(@config["IPOIB_MODE"])
-          "default"
+        case connection_config.ipoib_mode.name
```

Comment:
If you are expecting only two branches, what about using an `if`?

Reply:
makes sense, it is relict of old way where all types are managed manually.